src/elm/Pile.elm

```diff
@@ -0,0 +1,190 @@
+module Pile exposing (..)
+
+import Draggable
+import Draggable.Events
+import Html exposing (Html)
+import Html.Attributes
+import Html.Events
+import Math.Vector2 exposing (Vec2)
+
+
+type alias Id =
+    String
+
+
+type alias Card =
+    { id : Id
+    , position : Vec2
+    , cont : Html Msg
+    }
+
+
+dragCardBy : Vec2 -> Card -> Card
+dragCardBy delta card =
+    { card | position = card.position |> Math.Vector2.add delta }
+
+
+type alias CardPile =
+    { uid : Int
+    , activeCard : Maybe Card
+    , idleCards : List Card
+    }
+
+
+emptyPile : CardPile
+emptyPile =
+    CardPile 0 Nothing []
+
+
+addCard : ( Vec2, Html Msg ) -> CardPile -> CardPile
+addCard ( pos, cont ) ({ uid, idleCards } as pile) =
+    { pile
+        | uid = uid + 1
+        , idleCards = Card (String.fromInt uid) pos cont :: idleCards
+    }
+
+
+makeCardPile : List ( Vec2, Html Msg ) -> CardPile
+makeCardPile elements =
+    List.foldl addCard emptyPile elements
+
+
+allCards : CardPile -> List Card
+allCards { activeCard, idleCards } =
+    activeCard
+        |> Maybe.map (\c -> c :: idleCards)
+        |> Maybe.withDefault idleCards
+
+
+startDragging : Id -> CardPile -> CardPile
+startDragging id pack =
+    let
+        ( targetAsList, others ) =
+            List.partition (.id >> (==) id) (allCards pack)
```

Comment:
I tend to steer away from the function composition abstractions `>>` and `<<` but if the other way is too verbose, in this particular case, then it's fine.